Lib/profiling/sampling/_heatmap_assets/heatmap.js

```diff
@@ -331,6 +349,401 @@ document.addEventListener('click', e => {
     }
 });
 
+// ========================================
+// SPECIALIZATION VIEW TOGGLE
+// ========================================
+
+let specViewEnabled = false;
+
+/**
+ * Calculate heat color for given intensity (0-1)
+ * Hot spans (>30%) get warm orange, cold spans get dimmed gray
+ * @param {number} intensity - Value between 0 and 1
+ * @returns {string} rgba color string
+ */
+function calculateHeatColor(intensity) {
+    // Hot threshold: only spans with >30% of max samples get color
+    if (intensity > 0.3) {
+        // Normalize intensity above threshold to 0-1
+        const normalizedIntensity = (intensity - 0.3) / 0.7;
+        // Warm orange-red with increasing opacity for hotter spans
+        const alpha = 0.25 + normalizedIntensity * 0.35;  // 0.25 to 0.6
+        return `rgba(255, 100, 50, ${alpha})`;
```

Comment:
Maybe we can read from a CSS variable instead so we don't have to hardcode this/can reuse colors we already have defined?